From 1b41d4e65a5dc573b591d6df65b92939df273e69 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Fri, 5 Sep 2014 20:05:40 +0100
Subject: [PATCH 055/118] daemon: Check return value of end_stringsbuf.

Found by Coverity.

(cherry picked from commit d82cf08ce12ffd86e9f1588881dbe3f0ffd12294)
---
 daemon/lvm-filter.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/daemon/lvm-filter.c b/daemon/lvm-filter.c
index 2b9d933..d887947 100644
--- a/daemon/lvm-filter.c
+++ b/daemon/lvm-filter.c
@@ -280,8 +280,11 @@ make_filter_strings (char *const *devices)
   if (add_string (&ret, "r|.*|") == -1)
     goto error;
 
-  end_stringsbuf (&ret);
+  if (end_stringsbuf (&ret) == -1)
+    goto error;
+
   return ret.argv;
+
 error:
   if (ret.argv)
     free_stringslen (ret.argv, ret.size);
-- 
2.7.4

